src/plugins/imagery/components/Compass/Compass.vue

```diff
@@ -56,64 +55,47 @@ export default {
         CompassRose
     },
     props: {
-        containerWidth: {
-            type: Number,
+        compassRoseSizingClasses: {
+            type: String,
             required: true
         },
-        containerHeight: {
-            type: Number,
-            required: true
-        },
-        naturalAspectRatio: {
-            type: Number,
+        image: {
+            type: Object,
             required: true
         },
-        image: {
+        sizedImageDimensions: {
             type: Object,
             required: true
         },
-        lockCompass: {
+        testMode: {
             type: Boolean,
-            required: true
+            required: false,
+            default() {
+                return false
+            }
         }
     },
     computed: {
-        sizedImageDimensions() {
-            let sizedImageDimensions = {};
-            if ((this.containerWidth / this.containerHeight) > this.naturalAspectRatio) {
-                // container is wider than image
-                sizedImageDimensions.width = this.containerHeight * this.naturalAspectRatio;
-                sizedImageDimensions.height = this.containerHeight;
-            } else {
-                // container is taller than image
-                sizedImageDimensions.width = this.containerWidth;
-                sizedImageDimensions.height = this.containerWidth * this.naturalAspectRatio;
-            }
-
-            return sizedImageDimensions;
-        },
         hasCameraFieldOfView() {
             return this.cameraPan !== undefined && this.cameraAngleOfView > 0;
         },
         // horizontal rotation from north in degrees
         heading() {
-            return this.image.heading;
+            return this.testMode? this.getRandomAngle : this.image.heading;
```

Comment:
I think @nikhilmandlik  has [added some test data to example imagery](https://github.com/nasa/openmct/pull/3999). Maybe we could use that for testing instead of hard-coding it here?

Reply:
Yes, wasn't aware that @nikhilmandlik had already done that. I'll coordinate with him.

Reply:
Fixed